generators/beagle/src/main/scala/BeaglePeriphery.scala

```diff
@@ -192,42 +265,141 @@ trait HasPeripheryBeagleBundle {
+  val bh_clk_mux_rst = ResetCatchAndSync(bh_clk_mux.io.clockOut, reset.asBool)
+
+  val rs_clk_mux = testchipip.ClockMutexMux(single_clks ++ diff_clks)
+  rs_clk_mux.io.sel := rs_clk_sel
+  rs_clk_mux.io.resetAsync := rst_async
+  val rs_clk_mux_out = rs_clk_mux.io.clockOut
+  val rs_clk_mux_rst = ResetCatchAndSync(rs_clk_mux.io.clockOut, reset.asBool)
+
+  // setup clock dividers
+
+  val bh_clk_divider = withClockAndReset(bh_clk_mux_out, bh_clk_mux_rst) {
+    Module(new testchipip.ClockDivider(outer.scrParams.bhClkDivBits))
+  }
+  bh_clk_divider.io.divisor := scr_mod.io.bh_clk_divisor
+  val bh_clk = bh_clk_divider.io.clockOut
+  val bh_rst = ResetCatchAndSync(bh_clk, bh_clk_mux_rst)
+
+  val rs_clk_divider = withClockAndReset(rs_clk_mux_out, rs_clk_mux_rst) {
+    Module(new testchipip.ClockDivider(outer.scrParams.rsClkDivBits))
+  }
+  rs_clk_divider.io.divisor := scr_mod.io.rs_clk_divisor
+  val rs_clk = rs_clk_divider.io.clockOut
+  val rs_rst = ResetCatchAndSync(rs_clk, rs_clk_mux_rst)
+
+  val bh_clk_out_divider = withClockAndReset(bh_clk, bh_rst) {
+    Module(new testchipip.ClockDivider(outer.scrParams.bhClkDivBits))
+  }
+  bh_clk_out_divider.io.divisor := scr_mod.io.bh_clk_out_divisor
+  val bh_clk_div2_out = bh_clk_out_divider.io.clockOut
+
+  val rs_clk_out_divider = withClockAndReset(rs_clk, rs_rst) {
+    Module(new testchipip.ClockDivider(outer.scrParams.rsClkDivBits))
+  }
+  rs_clk_out_divider.io.divisor := scr_mod.io.rs_clk_out_divisor
+  val rs_clk_div2_out = rs_clk_out_divider.io.clockOut
+
+  val lbwif_clk_divider = withClockAndReset(uncore_clk, uncore_rst) {
+    Module(new testchipip.ClockDivider(outer.scrParams.lbwifClkDivBits))
+  }
+  lbwif_clk_divider.io.divisor := scr_mod.io.lbwif_clk_divisor
+  val lbwif_clk = lbwif_clk_divider.io.clockOut
+  val lbwif_rst = ResetCatchAndSync(lbwif_clk, uncore_rst)
+
+  // setup lbwif
+  lbwif_serial <> outer.lbwif.module.io.ser
   Seq(outer.lbwif, outer.lbwifCrossingSource, outer.lbwifCrossingSink).foreach { m =>
     m.module.clock := lbwif_clk
     m.module.reset := lbwif_rst
   }
 
+  // pass out clocks
+  uncore_clk_out := uncore_clk
+  bh_clk_out     := bh_clk_div2_out
+  rs_clk_out     := rs_clk_div2_out
+  lbwif_clk_out  := lbwif_clk
+
   // switch from lbwif and hbwif
-  outer.switcher.module.io.sel := outer.scr.module.io.switcher_sel
+  outer.switcher.module.io.sel := scr_mod.io.switcher_sel
 
   // setup hbwif
-  outer.hbwif.module.hbwifResets := outer.scr.module.io.hbwif_rsts
-  outer.hbwif.module.resetAsync := rst_async
-  outer.hbwif.module.hbwifRefClocks := hbwif_clks
-  hbwif_tx <> outer.hbwif.module.tx
-  hbwif_rx <> outer.hbwif.module.rx
-
-  // other
-  val clk_sel = outer.scr.module.io.clk_sel
-
-  outer.scr.module.io.boot := boot
-  outer.scr.module.io.rst_async := rst_async
+  val hbwif_mod = outer.hbwif.module
+  hbwif_mod.hbwifResets := scr_mod.io.hbwif_rsts
+  hbwif_mod.resetAsync := rst_async
+  hbwif_mod.hbwifRefClocks := hbwif_clks
```

Comment:
Did these get setup here? I didn't see it.

Reply:
These are setup from the MMIO and the clocks are "clock_rx"'ed at the ChipTop and passed into here.